tap_shopify/streams/products.py

```diff
@@ -186,20 +280,43 @@ class Products(Stream):
         }
     """
 
+    access_scopes_query = """
+        query CheckAppAccessScopes {
+        appInstallation {
+            accessScopes {
+            handle
+            }
+        }
+        }
+    """
+
     @shopify_error_handling
     def _call_api(self, query, variables):
         """
         Generalized method for making API calls with a GraphQL client.
+
+        Args:
+            query: The GraphQL query to execute
+            variables: Variables to pass to the query
+            error_handler: Optional function to handle errors (defaults to self._handle_error)
         """
         shopify.ShopifyResource.activate_session(Context.shopify_graphql_session)
         gql_client = shopify.GraphQL()
         response = gql_client.execute(query, variables)
         result = json.loads(response)
         shopify.ShopifyResource.activate_session(Context.shopify_rest_session)
         if result.get("errors"):
-            raise Exception(result['errors'])
+            error_handler = error_handler or self._handle_error
+            error_handler(result['errors'])
```

Comment:
Is this `error_handler` code something you added and then rolled back? Don't see it defined anywhere

Reply:
@hsyyid Good catch, it was exactly that. Just fixed.